### PR TITLE
C fraction development

### PR DESCRIPTION
**Pull request by BlackCorbeau**
This is all worked program for class Fraction.

### CODE REVIEW

**Comment by Roksyh** [comment]
:+1

**Comment by BlackCorbeau** [comment]
👍

**Comment by NiTro005** [comment]
В целом все хорошо 👍

File: CFraction/CFraction/CFraction.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+int Fraction::get_denuminator()
+{
+	return _denominator;
+}
+
+void Fraction::print()
+{
+	std::cout << _numerator << '/' << _denominator << '\n';
+}
+void Fraction::Do_easier()
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp]: -> simplify

File: CFraction/CFraction/CFraction.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
+#include "CFraction.h"
+#include <string>
+#include <iostream>
+
+
+Fraction::Fraction()
+{
+	_numerator = 0;
+	_denominator = 0;
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp]: Maybe, `_denominator = 1` is better?

File: CFraction/CFraction/CFraction.cpp [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+void Fraction::print()
+{
+	std::cout << _numerator << '/' << _denominator << '\n';
+}
+void Fraction::Do_easier()
+{
+	int m = _denominator;
+	int n = _numerator;
+	int r = m % n;
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp]: `5/2`: 
```cpp
m = 2
n = 5
r = m % n = 2 % 5 = 2
```
so, loop will be processed, but `5/2` is already simplified enough

File: CFraction/CFraction/CFraction.cpp (hunk 2 above)
Comment (VerySweetBread) [reply CFraction/CFraction/CFraction.cpp]: Do simplification in constructors

File: CFraction/CFraction/CFraction.cpp [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+{
+	if (this != &fr)
+	{
+		_denominator = fr._denominator;
+		_numerator = fr._numerator;
+	}
+	else { return *this; }
+}
+
+bool Fraction::operator!=(const Fraction& fr)
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:84]: You don't have simplification in constructors, that's why all operator overrides are incorrect: `1/2` = `2/4`

File: CFraction/CFraction/CFraction.cpp [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+{
+	_numerator = F1._numerator;
+	_denominator = F1._denominator;
+}
+Fraction::Fraction(int numerator, int denumirator)
+{
+	_numerator = numerator;
+	_denominator = denumirator;
+}
+//Fraction::Fraction(float num) //��������
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:37]: ?

File: CFraction/CFraction/CFraction.cpp [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+	}
+	else { return *this; }
+}
+
+bool Fraction::operator!=(const Fraction& fr)
+{
+	return (_denominator != fr._denominator && _numerator != fr._numerator);
+}
+
+bool Fraction::operator>=(const Fraction& fr)
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:89]: `3/4` >= `1/2`?
`4 ≠ 2`: `4 > 2`

File: CFraction/CFraction/main.cpp [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#include <iostream>
+#include "CFraction.h"
+
+int main()
+{
+	Fraction fr(5, 4);
+	Fraction fr2(8, 20);
+	Fraction Fr3;
```
Comment (VerySweetBread) [inline CFraction/CFraction/main.cpp]: Do not name objects starts with capital letter — only classes

File: CFraction/CFraction/CFraction.cpp [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+int Fraction::get_numerator()
+{
+	return _numerator;
+}
+int Fraction::get_denuminator()
+{
+	return _denominator;
+}
+
+void Fraction::print()
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:68]: You already have `<<` overload, why did you create argumentless `print`?

File: CFraction/CFraction/CFraction.cpp [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+Fraction Fraction::operator=(const Fraction& fr)
+{
+	if (this != &fr)
+	{
+		_denominator = fr._denominator;
+		_numerator = fr._numerator;
+	}
+	else { return *this; }
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:81]: Hint: you can don't wrap one line with `{}`:
```cpp
else return *this;
```

File: CFraction/CFraction/CFraction.cpp [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+{
+	Fraction Res(*this);
+	if (Res._denominator == fr1._denominator)
+	{
+		Res._numerator += fr1._numerator;
+	}
+	else
+	{
+		Res._numerator = (Res._numerator * fr1._denominator) + (fr1._numerator * Res._denominator);
+		Res._denominator = Res._denominator * fr1._denominator;
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:116]:
```cpp
Res._denominator *= fr1._denominator;
```

File: CFraction/CFraction/CFraction.cpp [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+	}
+	else
+	{
+		_numerator = (_numerator * fr1._denominator) - (fr1._numerator * _denominator);
+		_denominator = _denominator * fr1._denominator;
+	}
+	this->Do_easier();
+}
+
+Fraction Fraction::operator*(const Fraction& fr)
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:167]: `2/1`*`1/2`=`2/2` — do simplification here too

**Review by VerySweetBread — CHANGES_REQUESTED** [review]

File: CFraction/CFraction/CFraction.cpp (hunk 1 above)
Comment (BlackCorbeau) [reply CFraction/CFraction/CFraction.cpp]: ?

Comment (VerySweetBread) [reply CFraction/CFraction/CFraction.cpp]: > ?
> 

Rename this function to simplify, what the fuck is "do_easier"?

File: CFraction/CFraction/CFraction.cpp (hunk 5 above)
Comment (BlackCorbeau) [reply CFraction/CFraction/CFraction.cpp:37]: Why you dislike this two strings?

File: CFraction/CFraction/CFraction.cpp (hunk 8 above)
Comment (BlackCorbeau) [reply CFraction/CFraction/CFraction.cpp:68]: I forgot to delete it after I finished)

File: CFraction/CFraction/CFraction.cpp (hunk 11 above)
Comment (BlackCorbeau) [reply CFraction/CFraction/CFraction.cpp:167]: I already wrote siplification in this function)

File: CFraction/CFraction/CFraction.cpp (hunk 5 above)
Comment (VerySweetBread) [reply CFraction/CFraction/CFraction.cpp:37]: Äîïèñàòü

**Comment by BlackCorbeau** [comment]
@UsovaMA, Доброго времени суток, посмотрите пожалуйста pull requst.

File: CFraction/CFraction/CFraction.cpp (hunk 8 above)
Comment (VerySweetBread) [reply CFraction/CFraction/CFraction.cpp:68]: Still forget

File: CFraction/CFraction/CFraction.cpp [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+bool Fraction::operator!=(const Fraction& fr)
+{
+	return (_denominator != fr._denominator && _numerator != fr._numerator);
+}
+
+bool Fraction::operator>=(const Fraction& fr)
+{
+	return ((_denominator == fr._denominator && _numerator >= fr._numerator) || (_denominator <= fr._denominator));
```
Comment (VerySweetBread) [inline CFraction/CFraction/CFraction.cpp:91]: Still wrong

**Review by VerySweetBread — CHANGES_REQUESTED** [review]

**Pull request closed without merging**